### PR TITLE
chore: Add capabilities matrix

### PR DESCRIPTION
**Pull request by deanesmith**
High level guidance across Kubernetes providers of what key jx features work and what do not

Signed-off-by: deanesmith <dsmith@cloudbees.com>

### CODE REVIEW

**Comment by ankitm123** [comment]
What about the case where vault works in EKS 1.15, and does not work in EKS 1.16 (the vault operator chart had to be updated for it to be able to boot) - how would that be represented? Vault is just one example, there might be more ...

File: content/en/about/capabilities.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+## Jenkins X Capabilities Matrix
+
+| Capability | GKE | EKS | AKS | OpenShift 3.11 |
+| --- | :---:| :---: | :---: | :---: |
+| **Terraform scripts for cluster creation** | Yes | Yes | No | No |
+| **`jx boot` for installation** | Yes | Yes | No | Yes |
+| **Vault for secret storage** | Yes | Yes | No | Self Provisioned |
+| **Nexus** | Yes | Yes | No | Yes |
```
Comment (hferentschik) [inline content/en/about/capabilities.md:16]: Why is Nexus worth mentioning? That seems to me a bit random and too specific on this level of abstractio.

**Comment by hferentschik** [comment]
> What about the case where vault works in EKS 1.15, and does not work in EKS 1.16 (the vault operator chart had to be updated for it to be able to boot) - how would that be represented? Vault is just one example, there might be more ...

I feel the same way. The format of this table is in itself limiting, but I guess there is no easy way to depict the complexity of things. It might be better than nothing.

> Things can be version specific and platform specific at the same time, and that's where we will have issues with in this representation.

+1. Also, things might not be Kubernetes (cloud provider) specific at all like the SCM providers. 

I am wondering whether AKS should be even in the table, because we effectively have no AKS.

File: content/en/docs/install-setup/installing/boot/clouds/_index.md [hunk 2]
```diff
@@ -19,24 +19,6 @@ However if you try doing any significant work with Kubernetes particularly with
 
 This section of the documentation will try guide you through the best way to use Jenkins X on different cloud providers.
 
-## Cloud Provider Support Maturity
+Refer to the [Jenkins X Capabilities Matrix](/about/capabilities) for cloud provider support maturity.
```
Comment (hferentschik) [inline content/en/docs/install-setup/installing/boot/clouds/_index.md:22]: 👍

File: content/en/about/capabilities.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+title: Jenkins X Capabilities Matrix
+linktitle: Capabilities Matrix
+description: Matrix that describes the current state of Jenkins X capabilities for major Kubernetes platforms
+type: docs
+weight: 20
+---
+
+## Jenkins X Capabilities Matrix
+
+| Capability | GKE | EKS | OpenShift 3.11 |
```
Comment (hferentschik) [inline content/en/about/capabilities.md:11]: 👍

**Pull request merged**